roles/dcos_agent/tasks/main.yml

```diff
@@ -26,7 +26,7 @@
   shell: |
     set -o pipefail
     curl {{ dcos['config']['bootstrap_url'] }}/{{ dcos_version_specifier }}/genconf/serve/package_lists/$(curl {{ dcos['config']['bootstrap_url'] }}/{{ dcos_version_specifier }}/genconf/serve/cluster-package-list.latest).package_list.json | \
-    jq -r '.[] | select(.|test("dcos-config--setup"))' | sed s/dcos-config--//
+    jq '.' | grep dcos-config--setup | sed s/\"// | sed s/\",// | sed s/dcos-config--// | sed s/\ \ //
```

Comment:
The problem is that jq in flatcar is incomplete:
`jq: error (at <stdin>:0): jq was compiled without ONIGURUMA regex libary. match/test/sub and related functions are not available.`
As such we reworked the command until it worked with the features available in flatcar

Reply:
ah ok I see. Thanks for explaining

Reply:
so i'm still not completely fine with this part. I'm concerned that this stripped down jq could cause other problems. As the official jq is an dcos dependency I'd rather tent to have ansible overwriting the shipped jq version with the latest jq e.g. https://github.com/stedolan/jq/releases/download/jq-1.6/jq-linux64

Reply:
looking at https://github.com/flatcar-linux/manifest/releases/tag/v2513.0.0 
> Enable regex support for jq

Reply:
ok 2513 is definitly not an option Linux 5 is definitly unexpected.

```
Jul 15 16:36:07 ip-172-31-61-214.ec2.internal useradd[3314]: new user: name=dcos_mesos_dns, UID=476, GID=476, home=/opt/mesosphere, shell=/sbin/nologin
Jul 15 16:36:07 ip-172-31-61-214.ec2.internal pkgpanda[2924]: systemctl: symbol lookup error: /usr/lib/systemd/libsystemd-shared-243.so: undefined symbol: seccomp_api_get
Jul 15 16:36:07 ip-172-31-61-214.ec2.internal pkgpanda[2924]: ERROR: Command '['systemctl', 'daemon-reload']' returned non-zero exit status 127.
Jul 15 16:36:07 ip-172-31-61-214.ec2.internal systemd[1]: dcos-setup.service: Main process exited, code=exited, status=1/FAILURE
Jul 15 16:36:07 ip-172-31-61-214.ec2.internal systemd[1]: dcos-setup.service: Failed with result 'exit-code'.
Jul 15 16:36:07 ip-172-31-61-214.ec2.internal systemd[1]: Failed to start Pkgpanda: Specialize DC/OS for this host..
```

Reply:
so the path on a flatcar system looks like this:

```
/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/opt/bin
```

system jq is in /usr/bin but we cannot simply put another jq version somewhere else as all of these destinations are in RO filesystem.

Why is this a problem?:
So jq is not only used in ansible but is part of a lot of scripts and tools of DC/OS. Its an official requirement and its definitly possible that in some cases Regex is needed.

Without replacing jq with a proper version we expect I don't think this could be expected as safe to use.

Even flatcar enabled the regex support in later versions: see above. But those versions won't work out of the box as they are on Linux 5 which is definitly unexpected by DC/OS and leads to issues during install.

Reply:
I don't think the stripped down jq is a problem in practice. Up until 1.13 DC/OS officially supported CoreOS which also only had the restricted jq version and I doubt that much has changed in regards to jq usage.
Point in case: We've been running DC/OS 2.0 on CoreOS (although an older version) on a pre-production and several temporary test clusters for months without any problems related to jq.

Reply:
So i've just checked our docs. Even if its not too easy to find that list we definitely are expecting a fully usable jq as of https://github.com/stedolan/jq/wiki/Installation

So the quite obvious parts of DC/OS using `jq` are **currently** not using regex features. However that being said we still expect a jq with regex features which means that additional packages, scripts, debugging, guides or migrations might use it which then leads to errors.

No test will cover these cases. And especially migration scripts could lead to huge issues during future upgrade.

If we publish even best effort support for flatcar we must cover this issue avoiding impact on customer systems.

> Point in case: We've been running DC/OS 2.0 on CoreOS (although an older version) on a pre-production and several temporary test clusters for months without any problems related to jq.

Thats fine but there is no guarantee that this will be the case in the future or doing special operations ( don't think there is something like this yet, but I just checked a few sources for about 10 min ). In any case **there is no guarantee that this will always work**

Every automated testing is always done with full featured jq.


So I'd like to focus on solving this issue. We definitly need to place a fully jq in the path in favor of the systems jq. As I'm not an flatcar expert you might have an good idea how that could be solved?

Reply:
That is not an easy problem. As you noticed all the normal candidates are on RO filesystems. The flatcar way would be to put it in `/opt/bin`, but that does not override the system `jq`. We could still do that and then add files to `/etc/profile.d` and `/etc/bash/bashrc.d` that override the `PATH` so that `/opt/bin` is in first place. 

But I see one other problem: `/opt/mesosphere/environment` sets the path to `PATH=/opt/mesosphere/bin:/usr/bin:/bin:/sbin`. So I think that means the dcos services would not find our jq.

Reply:
Ok I've played around with different options.

Doing a bind mount into /usr/local/sbin which takes precedence
```
- name: Mount and bind a volume
  mount:
    path: /usr/local/sbin
    src: /opt/sbin
    opts: bind
    state: mounted
    fstype: none
```

This is working for interactive shells. But being problematic for ansible as the non-interactive shell has a pretty stripped down $PATH variable.

Now we could add a `source /etc/profile` into the ansible shell but it doesn't really solve the problem. To add the /usr/local/sbin also to non-interactive shells we probably have to change it in login.defs

An option which is IMO also not a good idea would be remount and override. I wouldn't really want to go that route.